controller/user_service_test.go

```diff
@@ -16,6 +24,35 @@ var (
 	ptb = ptr.Bool
 )
 
+type UserServiceControllerTestSuite struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestUserServiceController(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &UserServiceControllerTestSuite{})
+}
+
+func (s *UserServiceControllerTestSuite) SecuredController(identity repository.Identity) (*goa.Service, *UserServiceController) {
+	svc := testsupport.ServiceAsUser("User-Service", identity)
+	controller := NewUserServiceController(svc, s.Application)
+	return svc, controller
+}
+
+func (s *UserServiceControllerTestSuite) UnsecuredController() (*goa.Service, *UserinfoController) {
+	svc := goa.New("Userinfo-Service")
+	controller := NewUserinfoController(svc, s.Application, testtoken.TokenManager)
+	return svc, controller
+}
+
+func (s *UserServiceControllerTestSuite) TestShowOK() {
+	identity, err := testsupport.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "userTestShowUserOK"+uuid.NewV4().String())
+	require.Nil(s.T(), err)
+
+	svc, ctrl := s.SecuredController(identity)
+	test.ShowUserServiceInternalServerError(s.T(), svc.Context, svc, ctrl)
```

Comment:
It doesn't seem to test ShowOK :)

Reply:
Oops.. I've fixed the function name now.

Reply:
OK. But what about ShowOK test? We would have caught the bug if we had that but test in the first place, wouldn't we?

Reply:
We can't easily get an "OK" result from the controller.. we have extensive tests for the TenantService in `tenant_whitebox_test.go`, however it is testing an internally-created tenant service for which the `rest.HttpDoer` is replaced with a dummy object.  We don't have the mechanism to easily do this from the controller.

Reply:
You can use `factory.WithTenantService(serviceMock)`
See example https://github.com/fabric8-services/fabric8-auth/blob/master/controller/users_blackbox_test.go#L57

Reply:
I'm updating the test with this.

Reply:
All looks good now.